version: "3.8"
services:
  sa_appointment_postgres:
    image: postgres:17
    environment:
      POSTGRES_DB: sa_appointment
      POSTGRES_USER: myuser
      POSTGRES_PASSWORD: mypassword
    ports:
      - "5433:5432"
    volumes:
      - ./volume/postgres:/var/lib/postgresql/data
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U myuser -d sa_appointment"]
      interval: 3s
      timeout: 2s
      retries: 20
    networks:
      - default

  sa_appointment:
    image: sa_appointment:latest
    build:
      context: .
      dockerfile: Dockerfile
    environment:
      - DB_HOST=sa_appointment_postgres
      - DB_PORT=5432
      - USER_SERVICE_URL=http://host.docker.internal:8000/api/user
    ports:
      - "8001:8001"
    depends_on:
      sa_appointment_postgres:
        condition: service_healthy
    networks:
      - default

networks:
  default:
    driver: bridge
